packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation_api_impls.dart

```diff
@@ -35,14 +37,56 @@ Iterable<NSKeyValueObservingOptionsEnumData>
   });
 }
 
+/// Handles initialization of Flutter APIs for the Foundation library.
+class FoundationFlutterApis {
+  /// Constructs a [FoundationFlutterApis].
+  ///
+  /// This should only be changed for testing purposes.
+  @visibleForTesting
+  FoundationFlutterApis({
+    this.binaryMessenger,
+    InstanceManager? instanceManager,
+  }) {
+    functionFlutterApi =
+        FunctionFlutterApiImpl(instanceManager: instanceManager);
+  }
+
+  /// Mutable instance containing all Flutter Apis for the Foundation library.
+  ///
+  /// This should only be changed for testing purposes.
+  static FoundationFlutterApis instance = FoundationFlutterApis();
+
+  /// Sends binary data across the Flutter platform barrier.
+  final BinaryMessenger? binaryMessenger;
+
+  bool _hasBeenSetUp = false;
+
+  /// Flutter Api for disposing functions.
+  @visibleForTesting
+  late final FunctionFlutterApiImpl functionFlutterApi;
```

Comment:
This was placed in the `FoundationFlutterApis` because `FoundationFlutterApis.instance.ensureSetUp();` is called inside `NSObject`. And `NSObject` will be the parent class of all objects.